python/ray/autoscaler/_private/monitor.py

```diff
@@ -134,17 +134,20 @@ class Monitor:
 
     def __init__(self,
                  redis_address,
+                 gcs_address,
                  autoscaling_config,
                  redis_password=None,
                  prefix_cluster_info=False,
                  monitor_ip=None,
                  stop_event: Optional[Event] = None):
-        # Initialize the Redis clients.
-        self.redis = ray._private.services.create_redis_client(
-            redis_address, password=redis_password)
-        (ip, port) = redis_address.split(":")
-        # Initialize the gcs stub for getting all node resource usage.
-        gcs_address = self.redis.get("GcsServerAddress").decode("utf-8")
+        if not use_gcs_for_bootstrap():
```

Comment:
Since we are checking the feature flag `use_gcs_for_bootstrap()` here anyway, it might be easier to accept just `address` as `Monitor.__init__()` parameter, instead of separate `redis_address` and `gcs_address`. See my fix for other instances of `Monitor` creation which needs to check `use_gcs_for_bootstrap()` to determine whether to set `address` to `redis_address` or `gcs_address` for `Monitor`.

Reply:
The reason I always add extra parameters instead of reusing the existing one and interpreting it differently is that I feel it might be more clear and also py test can catch the error easily.

In the end, we'll have a PR to stop passing redis address around, and when that is enabled, whenever when we use the redis address, it'll throw exception and it can help us identify the place where broken easier.

Reply:
I think you are right about this one. It seems like there are other components (like ray operator) are using this API. I'll go with your comment.